src/cephadm/cephadm.py

```diff
@@ -979,6 +981,133 @@ def get_tcmu_runner_container(self):
         tcmu_container.cname = self.get_container_name(desc='tcmu')
         return tcmu_container
 
+
+##################################
+
+
+class CephNvmeof(object):
+    """Defines a Ceph-Nvmeof container"""
+
+    daemon_type = 'nvmeof'
+    required_files = ['ceph-nvmeof.conf']
+    default_image = DEFAULT_NVMEOF_IMAGE
+
+    def __init__(self,
+                 ctx,
+                 fsid,
+                 daemon_id,
+                 config_json,
+                 image=DEFAULT_NVMEOF_IMAGE):
+        # type: (CephadmContext, str, Union[int, str], Dict, str) -> None
+        self.ctx = ctx
+        self.fsid = fsid
+        self.daemon_id = daemon_id
+        self.image = image
+
+        # config-json options
+        self.files = dict_get(config_json, 'files', {})
+
+        # validate the supplied args
+        self.validate()
+
+    @classmethod
+    def init(cls, ctx, fsid, daemon_id):
+        # type: (CephadmContext, str, Union[int, str]) -> CephNvmeof
+        return cls(ctx, fsid, daemon_id,
+                   fetch_configs(ctx), ctx.image)
+
+    @staticmethod
+    def get_container_mounts(data_dir: str) -> Dict[str, str]:
+        mounts = dict()
+        mounts[os.path.join(data_dir, 'config')] = '/etc/ceph/ceph.conf:z'
+        # mounts[os.path.join(data_dir, 'keyring')] = '/etc/ceph/keyring:z'
+        mounts['/etc/ceph/ceph.client.admin.keyring'] = '/etc/ceph/keyring:z'  # TODO: FIXME
```

Comment:
two things with this
1) Do we actually need the admin keying? Ideally we'd use one with only the required permissions while the admin keyring allows everything
2) There's no guarantee of an admin keyring being present in `/etc/ceph/` and in multicluster scenarios (which we technically try to support) it's not even guaranteed this keyring is for this cluster. If we DO need the admin keyring, we should probably provide it from the mgr module the same way we do with any other daemon keyrings.

Reply:
Definitely no need to employ the admin keyring.

`profile rbd` cap would be needed for SPDK and I think it could be used for the gateway as well (it should grant the necessary access to `nvmeof.state` objects in the same pool/namespace as is).  Having to manage one common NVMEoF key is simpler, but we could come up with a separate, much more restricted, key for the gateway along the lines of what @epuertat proposed.